src/main/java/net/minecraftforge/common/brewing/IBrewingRecipe.java

```diff
@@ -20,26 +20,32 @@
 package net.minecraftforge.common.brewing;
 
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeMod;
 
-public interface IBrewingRecipe {
+public interface IBrewingRecipe extends Recipe<IBrewingContainer>
+{
+    @Override
+    default boolean matches(final IBrewingContainer container, final Level level)
+    {
+        return isBase(container.base()) && isReagent(container.reagent());
+    }
 
-    /**
-     * Returns true is the passed ItemStack is an input for this recipe. "Input"
-     * being the item that goes in one of the three bottom slots of the brewing
-     * stand (e.g: water bottle)
-     */
-    boolean isInput(ItemStack input);
+    @Override
+    default boolean canCraftInDimensions(int p_43999_, int p_44000_)
```

Comment:
Readable parameter names.

Not needed, but still an interesting thought. Since you mentioned that this could be expanded to larger containers, the width and height determined here could be used to calculate if there are enough spaces for more than the specified number of brewing recipes.